src/socket/tcp.rs

```diff
@@ -749,6 +773,41 @@ impl<'a> Socket<'a> {
         }
     }
 
+    /// Return the local receive window scaling factor defined in [RFC 1323].
+    ///
+    /// The value will become constant after the connection is established.
+    /// It may be reset to 0 during the handshake if remote side does not support window scaling.
```

Comment:
I don't like that this API has a bidirectional data flow, where a value set by the consumer is read by the TCP/IP stack, but also then is updated by the TCP/IP stack. Is there any precedent (whether in smoltcp or in BSD TCP/IP) to have an option like this? I would there to be two API entry points, one to request an option, one to see if it was applied.

Reply:
In google's [gVisor userspace TCP/IP stack](https://github.com/google/gvisor/blob/ddfbb50ecc2d9b062d96b93a06aef86441e158d1/pkg/tcpip/transport/tcp/endpoint.go#L3077), they have similar options. I believe the issue is that, smoltcp has its default, and we want to override this default. That's why we have such a bidirectional data flow. Our TCP/IP stack reads the value (after connect/accept), and sets the value if no appropriate default value is set (before connect/accept).

As for the "two API entry points", I don't quite get that. Could you give some examples?